kernel/syscalls/kill.rs

```diff
@@ -0,0 +1,40 @@
+use crate::process::{current_process, signal::Signal, PId};
+use crate::result::Errno;
+use crate::result::Result;
+use crate::syscalls::SyscallHandler;
+
+impl<'a> SyscallHandler<'a> {
+    pub fn sys_kill(&self, pid: PId, sig: Signal) -> Result<isize> {
+        match pid.as_i32() {
+            pid if pid > 0 => {
+                if current_process().pid().as_i32() == pid {
+                    current_process().send_signal(sig);
+                    Ok(0)
+                } else {
+                    Err(Errno::ESRCH.into())
+                }
+            }
+
+            0 => {
+                current_process().process_group().lock().signal(sig);
+                Ok(0)
```

Comment:
Since kill(2) always return zero on success, how about adding a single `Ok(0)` at the end of this method instead?

```rust
match pid.as_i32() {
  /* ... */
}

Ok(0)

```